ideaforce/src/main/java/com/emc/ideaforce/controller/AdminController.java

```diff
@@ -44,9 +53,33 @@ public ModelAndView showAdminPage(Principal principal) {
 
     @RequestMapping(value="/approve/{id}", method = GET)
     @RolesAllowed(Utils.ADMIN_ROLE)
-    public String approveStory(Principal principal, @PathVariable String id) {
+    public String approveStory(@PathVariable String id) {
         commonService.setStoryApproved(id);
-        return "approved";
+        return APPROVE_CHALLENGE;
+    }
+
+    @RequestMapping(value="/addcomments/{id}", method = GET)
+    public ModelAndView commentsForStory(@PathVariable String id, ModelMap model) {
+        CommentDto commentDto = new CommentDto();
+        commentDto.setStoryId(id);
+        model.addAttribute("newcomment", commentDto);
+        return new ModelAndView(ADD_COMMENTS_VIEW, model);
+    }
+
+    @RequestMapping(value="/addcomments", method = POST)
+    public ModelAndView addCommentsForStory(@ModelAttribute("newcomment") CommentDto commentModel,
```

Comment:
We do not need these many apis, following would be ssufficient
`/viewcomments/{id}` -> `/review/{id}` (GET)
`/addcomments` -> `/review/{id}` (POST)
`/addcomments/{id}` -> `/review/{id}` (POST)
`/approve/{id}` -> `/review/{id}` (POST)

Restructuring `CommentDto` to `ReviewDto` and have the `reviewstatus` and `comments` inside that obj

Reply:
I am not able structure apis in this format. Need ```/addcomments/{id}``` for routing ui to comment page.